csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -88,11 +114,11 @@ internal async Task OpenAsync()
 
         protected internal HiveServer2TlsOption TlsOptions { get; set; } = HiveServer2TlsOption.Empty;
 
-        protected internal int HttpRequestTimeout { get; set; } = 30000;
+        protected internal int ConnectTimeoutMilliseconds { get; set; } = ConnectTimeoutMillisecondDefault;
 
-        protected abstract Task<TTransport> CreateTransportAsync();
+        protected abstract Task<TTransport> CreateTransportAsync(CancellationToken cancellationToken = default);
```

Comment:
It doesn't look like any of the implementations of `CreateTransportAsync` make use of the `CancellationToken`. Is this aspirational? `TTransport.Open` isn't called until `CreateProtocolAsync`.

Hmm... it looks like `TSocketTransport` itself isn't very async-friendly :(. It's doing DNS lookup and opening sockets in its constructors.